hugr-py/src/hugr/_ops.py

```diff
@@ -121,8 +121,13 @@ def __call__(self, tuple_: Wire) -> Command:
         return super().__call__(tuple_)
 
 
+class DfParentOp(Op, Protocol):
+    def input_types(self) -> tys.TypeRow: ...
+    def output_types(self) -> tys.TypeRow: ...
```

Comment:
These might be nicer as properties?

Reply:
I will be changing this in a follow up PR to be Op.signature() anyway so gonna leave it as is in this one